libp2p/RLPxHandshake.cpp

```diff
@@ -298,112 +325,163 @@ void RLPXHandshake::transition(boost::system::error_code _ech)
+                                                  << "@" << m_socket->remoteEndpoint() << ")";
+                                    m_nextState = Error;
+                                    transition();
+                                    return;
+                                }
+
+                                PacketType packetType =
+                                    frame[0] == 0x80 ? HelloPacket : (PacketType)frame[0];
+                                if (packetType != HelloPacket)
+                                {
+                                    LOG(m_logger)
+                                        << connectionDirectionString()
+                                        << "Invalid packet type. Expected: "
+                                        << packetTypeToString(HelloPacket)
+                                        << ", received: " << packetTypeToString(packetType) << " ("
+                                        << m_remote << "@" << m_socket->remoteEndpoint() << ")";
+                                    m_nextState = Error;
+                                    transition();
+                                    return;
+                                }
+
+                                LOG(m_logger) << connectionDirectionString()
+                                              << packetTypeToString(HelloPacket)
+                                              << " verified. Starting session with " << m_remote
+                                              << "@" << m_socket->remoteEndpoint();
+                                try
+                                {
+                                    RLP rlp(
+                                        frame.cropped(1), RLP::ThrowOnFail | RLP::FailIfTooSmall);
+                                    m_host->startPeerSession(m_remote, rlp, move(m_io), m_socket);
+                                }
+                                catch (std::exception const& _e)
+                                {
+                                    stringstream errorStream;
+                                    errorStream << connectionDirectionString() << "Handshake with "
+                                                << m_remote;
+                                    if (remoteSocketConnected())
+                                        errorStream << "@" << m_socket->remoteEndpoint();
+                                    errorStream << " causing an exception: " << _e.what();
+                                    LOG(m_errorLogger) << errorStream.str();
+                                    m_nextState = Error;
+                                    transition();
+                                }
+                            }
+                        });
+                }
+            });
     }
 }
+
+const char* RLPXHandshake::connectionDirectionString() const
+{
+    if (m_originated)
+        return "p2p.connect.egress: ";
```

Comment:
Maybe let's remove `p2p.connect.` here, as it should be clear what it's about from `rlpx` channel name ?

Reply:
Also I would make this function inline